From 24ac7f1e3e883609794dcae3521d30f757954786 Mon Sep 17 00:00:00 2001
From: danascape <danascape@gmail.com>
Date: Sun, 19 Feb 2023 03:47:00 +0530
Subject: [PATCH] security: selinux: Panic kernel before initialising

* This was performed intentionally as a task for Linux kernel Bug Fixing 2023
  (Spring).

dmesg:
[    0.120405] SELinux:  Initializing.
[    0.120549] Kernel panic - not syncing: SELinux: Kill the kernel
[    0.120799] CPU: 0 PID: 0 Comm: swapper/0 Not tainted 6.2.0-rc7-gc14b5ce65441 #8
[    0.121005] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.15.0-1 04/01/2014
[    0.121191] Call Trace:
[    0.121191]  <TASK>
[    0.121191]  dump_stack_lvl+0x38/0x4c
[    0.121191]  panic+0x106/0x286
[    0.121191]  selinux_init+0x1c/0x1c
[    0.121191]  initialize_lsm+0x2f/0x4c
[    0.121191]  security_init+0x3fd/0x41c
[    0.121191]  start_kernel+0x614/0x68d
[    0.121191]  secondary_startup_64_no_verify+0xe0/0xeb
[    0.121191]  </TASK>
[    0.121191] ---[ end Kernel panic - not syncing: SELinux: Kill the kernel ]---

Signed-off-by: danascape <danascape@gmail.com>
---
 security/selinux/hooks.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index 3c5be76a9199..7cb0d141a404 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -7332,6 +7332,7 @@ static struct security_hook_list selinux_hooks[] __lsm_ro_after_init = {
 static __init int selinux_init(void)
 {
 	pr_info("SELinux:  Initializing.\n");
+	panic("SELinux: Kill the kernel");
 
 	memset(&selinux_state, 0, sizeof(selinux_state));
 	enforcing_set(&selinux_state, selinux_enforcing_boot);
-- 
2.34.1

